modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -185,7 +185,27 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
                 repositoriesMetrics.throttleCounter().incrementBy(throttleCount, attributes);
                 repositoriesMetrics.throttleHistogram().record(throttleCount, attributes);
             }
-            repositoriesMetrics.httpRequestTimeInMicroHistogram().record(getHttpRequestTimeInMicros(request), attributes);
+            maybeRecordHttpRequestTime(request);
+        }
+
+        /**
+         * Used for APM style metrics to measure statics about performance. This is not for billing.
+         */
+        private void maybeRecordHttpRequestTime(Request<?> request) {
+            final List<TimingInfo> requestTimesIncludingRetries = request.getAWSRequestMetrics()
+                .getTimingInfo()
+                .getAllSubMeasurements(AWSRequestMetrics.Field.HttpRequestTime.name());
+            // It can be null if the request did not reach the server for some reason
+            if (requestTimesIncludingRetries == null) {
+                return;
+            }
```

Comment:
This is the main change that prevents a NPE being thrown when it gets iterated below. It's pretty straightforward. But let me know if you think a test is necessary.